tests/test_smoke.py

```diff
@@ -3012,6 +3012,88 @@ def terminate_replica(replica_id: int) -> str:
     run_one_test(test)
 
 
+@pytest.mark.gcp
+@pytest.mark.sky_serve
+def test_skyserve_base_ondemand_fallback():
+    name = _get_service_name()
+    test = Test(
+        f'test-skyserve-base-ondemand-fallback',
+        [
+            f'sky serve up -n {name} -y tests/skyserve/spot/base_ondemand_fallback.yaml',
+            _SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=2),
+            f'output=$(sky serve status {name});'
+            'echo "$output" | grep "GCP(\[Spot\]vCPU=2)" && '
+            'echo "$output" | grep "GCP(vCPU=2)";'
+        ],
+        _TEARDOWN_SERVICE.format(name=name),
+        timeout=20 * 60,
+    )
+    run_one_test(test)
+
+
+@pytest.mark.gcp
+@pytest.mark.sky_serve
+def test_skyserve_dynamic_ondemand_fallback():
+    name = _get_service_name()
+    zone = 'us-central1-a'
+
+    # Reference: test_spot_recovery_gcp
+    def terminate_replica(replica_id: int) -> str:
+        cluster_name = serve.generate_replica_cluster_name(name, replica_id)
+        query_cmd = (f'gcloud compute instances list --filter='
+                     f'"(labels.ray-cluster-name:{cluster_name})" '
+                     f'--zones={zone} --format="value(name)"')
+        return (f'gcloud compute instances delete --zone={zone}'
+                f' --quiet $({query_cmd})')
+
+    def _check_both_spot_on_demand_in_status(name: str) -> str:
+        return (f'output=$(sky serve status {name});'
+                'echo "$output" | grep "GCP(\[Spot\]vCPU=2)" && '
+                'echo "$output" | grep "GCP(vCPU=2)";')
```

Comment:
Actually that makes me think, should we print the `is_spot` column in our sky serve status table?

Reply:
The row will have `\[Spot\]` so I think we don't need a `is_spot` column